sql/catalog/lease: TestParallelLeaseAcquireWithImmediateRelease failed

sql/catalog/lease.TestParallelLeaseAcquireWithImmediateRelease [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20518038?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20518038?buildTab=artifacts#/) on master @ [bf7add615d6689dc7c738f83411c000851bdd1d7](https://github.com/cockroachdb/cockroach/commits/bf7add615d6689dc7c738f83411c000851bdd1d7):


```
=== RUN   TestParallelLeaseAcquireWithImmediateRelease
    test_log_scope.go:171: test logs captured to: /artifacts/tmp/_tmp/5632b52c1c67b61307e1d8a1ca2dd18d/logTestParallelLeaseAcquireWithImmediateRelease1626291813
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:154: automatically injected a shared process virtual cluster under test; see comment at top of test_server_shim.go for details.
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x27eeebc]

goroutine 10139 [running]:
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.(*descriptorVersionState).hasFixedExpiration(...)
	pkg/sql/catalog/lease/descriptor_version_state.go:143
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.(*Manager).purgeOldVersions(0x400d70b348, {0x617c570, 0x4005651d70}, 0x511c1c?, 0x6a, 0x0, 0x1)
	pkg/sql/catalog/lease/lease.go:1282 +0x30c
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.(*Manager).StartRefreshLeasesTask.func1.2({0x617c570, 0x4005651d70})
	pkg/sql/catalog/lease/lease.go:2029 +0x30c
github.com/cockroachdb/cockroach/pkg/sql/catalog/lease.(*Manager).StartRefreshLeasesTask.func1({0x617c570, 0x4005651d70})
	pkg/sql/catalog/lease/lease.go:2055 +0x3cc
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func1({0x617c570, 0x4005651d70})
	pkg/util/stop/stopper.go:440 +0x88
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 9521
	pkg/util/stop/stopper.go:438 +0x104
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestParallelLeaseAcquireWithImmediateRelease.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-54763

--- a/pkg/sql/catalog/lease/lease.go
+++ b/pkg/sql/catalog/lease/lease.go
@@ -1277,6 +1277,11 @@ func (m *Manager) purgeOldVersions(
 		}
 		// We encountered an error telling us to renew the lease.
 		newest := m.findNewest(id)
+		// It is possible that a concurrent drop / removal of this descriptor is
+		// occurring. If the newest version just doesn't exist, bail out.
+		if newest == nil {
+			break
+		}
 		// Assert this should never happen due to a fixed expiration, since the range
 		// feed is responsible for purging old versions and acquiring new versions.
 		if newest.hasFixedExpiration() {

